### DISCUSSION TITLE
How to Upgrade Home Assistant Created Using the Home Assistant  Core Script

### DISCUSSION THREAD

**Discussion by gcortes** [tteck/Proxmox#2461] [Q&A] [answered]
I need to upgrade my instance of Home Assistant that I created using the Core script. From reading the script and other research, it looks like I just need to do the following:

`cd /srv/homeassistant`
`source bin/activate`
`pip3 install --upgrade homeassistant`

Just pip was used in the script so I'm not sure pip3 is needed. Is this the correct procedure for the script install?

**Comment by tteck** [comment]
I'm not sure what you're asking.
If you want to update Home Assistant only, type the command `update` in the LXC console.

If you're wanting to update Python to 3.12 follow https://github.com/tteck/Proxmox/discussions/2458

**Reply by gcortes** [reply]
I want to upgrade Home Assistant from 2024.1.5 to 2024.2. What does update specifically do? Is there documentation for it? Did you mean apt update?

Sorry, I didn't read your second response before posting.

I moved the config files from /root/.homeassistant to another location so I can access the config via ssh without connecting as root if that makes any difference.

**Reply by tteck** [reply]
Moving the configuration folder without modifying the `homeassistant.service` will break the system. You should create a symlink instead.

**Reply by tteck** [reply]
Also, it will result in you not being able to utilize any of the [copy-data scripts](https://github.com/tteck/Proxmox/blob/main/misc/copy-data/README.md)

**Reply by tteck** [reply]
As for what executing `update` does, look at https://github.com/tteck/Proxmox/blob/main/ct/homeassistant-core.sh for the `update_script` function.

**Reply by gcortes** [reply]
I changed the homeassistant.service to point to the new location. Home Assistant runs fine. I took a look at the copy scripts. I'm not sure why I would need them. I do backups on own.  Given that I've made the changes I did, would I be OK with the steps I originally gave? I'm pretty sure I am, but I just wanted to double check. Again, given my changes, I may be on my own with this one. It may be time for a smoke test, after a backup of course. 
Thanks for your quick responses.

**Reply by tteck** [reply]
You _should_ be fine. And those [copy-data scripts](https://github.com/tteck/Proxmox/blob/main/misc/copy-data/README.md) come in handy when needing to quickly and safely transfer data between containers. Currently, updating to 2024.02.1 is going to show a warning about Python 3.11 support being removed.

![image](https://github.com/tteck/Proxmox/assets/21342670/4fe4c0d4-65d5-4a07-9cc8-ff7b65a6dba4)

The instructions at https://github.com/tteck/Proxmox/discussions/2458 will not apply to your setup, so yes, you'll "be on my own with this one."

**Reply by tteck** [reply]
Here ya go
```
systemctl stop homeassistant && source /srv/homeassistant/bin/activate && pip install --upgrade homeassistant && systemctl start homeassistant && exit
```

**Reply by gcortes** [reply]
Thanks for the command chain. I did a search on upgrading Python for Home Assistant. It looked like it could be a headache. I think I would be better off going back to the standard install of your script and using your tools. I still want to be able to remotely edit the config file using sftp and my own user id. You mentioned earlier creating a symlink to the Home Assistant directory to gain access without being root. Would just doing a chown on /root/.homeassistant work just as well? I don't see how creating a symlink gives you permission. Then again, I don't use symlinks much so may be missing something.

**Reply by tteck** [reply]
You can create a symbolic link from `/root/.homeassistant` to e.g., `/opt/config`
```
ln -s /opt/config /root/.homeassistant
```
If you intend to use a different user to access the contents of `/opt/config`, you'll need to ensure that user has the necessary permissions to read/write files in that directory and its subdirectories. 
```
# Change ownership to a different user and group
chown -R username:groupname /opt/config

# Adjust permissions (e.g., giving read/write access to the user)
chmod -R u+rw /opt/config
```
The new user will be able to access the contents of `/opt/config` through `/root/.homeassistant`. However, they won't have direct permission to modify the files within `/root/.homeassistant` due to its restricted permissions. You would need to modify the files within `/opt/config`, and those changes would be reflected in `/root/.homeassistant` due to the symbolic link.

**Reply by tteck** [reply]
Home Assistant is created/run with user `root`

![image](https://github.com/tteck/Proxmox/assets/21342670/a80af683-e927-43be-96d6-abb1f415cec6)

**Reply by tteck** [reply]
To execute any of the supplemental scripts, including the `update` command, would need to be user `root`

**Reply by tteck** [reply]
In my opinion, File Browser is the most convenient way to edit the config files in Home Assistant.

**Reply by gcortes** [reply]
I'm a long time Sublime user and also use ssh with the JetBrains IDE's. I watched a video on File Browser and see that you have a script for it so I may give it a try.

I did a chown on root/.homeassistant and can now access it via sftp. I'm failing to see why I would need a symlink to the directory when I can just access it directly. Is it a best practice type of thing?

**Reply by gcortes** [reply]
I tried copying the home assistant data back to /root/.homeassistant and changing the homeassistant.service. I then did deamon refresh and restarted home assistant. It came up without complaint, but then hung. I changed the service back and it ran normally again.

Rather than trying to fix it, I want to move on to creating a new ha core LXC to address the Python version issue. From looking at your copy script for core to core, it looks like I just need to change "CORE_PATH=/root/.homeassistant" to point to where I have the ha data.  Does that sound right? I could just try it, but I'm in no hurry so I thought I would check first.

**Comment by tteck** [comment, marked as answer]
![Screenshot 2024-02-09 2 04 27 PM](https://github.com/tteck/Proxmox/assets/21342670/ca031482-420c-4207-8765-0b52bae65bee)
